### PR TITLE
02-client: merge misbehavior & header interfaces

### PR DESCRIPTION
**Pull request by seantking**
## Description

I've had to add `GetHeight()` temporarily to the light client implementations to satisfy the compiler. Once we complete https://github.com/cosmos/ibc-go/issues/594 we can remove them.

closes: #875 
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/light-clients/06-solomachine/types/misbehaviour.go [hunk 1]
```diff
@@ -70,3 +70,9 @@ func (sd SignatureAndData) ValidateBasic() error {
 
 	return nil
 }
+
+// TODO: Remove GetHeight()
+// GetHeight implements the curret exported.Header interface, to be updated
+func (misbehaviour Misbehaviour) GetHeight() exported.Height {
```
Comment (seantking) [inline modules/light-clients/06-solomachine/types/misbehaviour.go:76]: We can remove this later. See PR description

File: modules/core/exported/client.go [hunk 2]
```diff
@@ -194,20 +194,12 @@ type ConsensusState interface {
 	ValidateBasic() error
 }
 
-// Misbehaviour defines counterparty misbehaviour for a specific consensus type
-type Misbehaviour interface {
+// ClientMessage is an interface used for Header and Misbehaviour implementations
```
Comment (seantking) [inline modules/core/exported/client.go:197]: @AdityaSripal suggestion would be welcome for the comment/description here.

**Comment by seantking** [comment]
The reason there are so many commits in the UI commit diff is just that I branch off of main rather than the feature branch by mistake. I have since [merged in the latest chains from main](https://github.com/cosmos/ibc-go/pull/1106). Github cache is just not updating the UI unfortunately. There are no changes outside of my commit and we will squash the branch anyway.

File: modules/core/exported/client.go (hunk 2 above)
Comment (colin-axner) [reply modules/core/exported/client.go:197]: Maybe something along the lines:
```
ClientMessage is an interface used to update an IBC client. The update may be done by a single header, a batch of headers, misbehaviour, or any type which when verified produces a change to state of the IBC client

**Review by colin-axner — APPROVED** [review]
Great work!

File: modules/core/02-client/types/encoding.go [hunk 3]
```diff
@@ -86,29 +86,29 @@ func UnmarshalConsensusState(cdc codec.BinaryCodec, bz []byte) (exported.Consens
...
-// MarshalHeader protobuf serializes a Header interface
-func MarshalHeader(cdc codec.BinaryCodec, h exported.Header) ([]byte, error) {
-	return cdc.MarshalInterface(h)
+// MarshalClientMessage protobuf serializes a ClientMessage interface
+func MarshalClientMessage(cdc codec.BinaryCodec, clientMessage exported.ClientMessage) ([]byte, error) {
+	return cdc.MarshalInterface(clientMessage)
 }
 
-// MustMarshalHeader attempts to encode a Header object and returns the
+// MustMarshalClientMessage attempts to encode a Header object and returns the
```
Comment (damiannolan) [inline modules/core/02-client/types/encoding.go:94]:
```suggestion
// MustMarshalClientMessage attempts to encode a ClientMessage object and returns the
```

**Review by damiannolan — APPROVED** [review]
Nice, LGTM! I can rebase the solomachine PRs and remove all the todos on the `ClientMessage` interface then! :)

**Pull request merged**